File: typescript/queryBaterries/queryBatteries.ts
import { PublicAPIClient as APIClient } from "@novum-batteries/cloud-api-client";

const apiClient = new APIClient();

(async () => {
  try {
    await apiClient.login("YOUR_EMAIL", "YOUR_PASSWORD");

    // In order to query your battery you have 3 optional arguments to use: filters, options and fields as arguments
    const filters = { name: "MY_BATTERY" }; //  any {[key: string]: any;}
    const options = { limit: 100 }; //{ sort?: { [key: string]: number },limit?: number,offset?: number}
    const fields = {}; // any {[key: string]: number};
    const batteries = await apiClient.getBatteries(filters, options, fields);

    console.log(batteries);
  } catch (e: any) {
    console.error(e.details);
  }
})();

File: python/queryBatteries/queryBatteries.py
from novum_api_client.client import NovumAPIClient
from novum_api_client.api_type import  ApiOptions

api = NovumAPIClient()

try:
    login = api.login("YOUR_EMAIL", "YOUR_PASSWORD")
    # This function allow you to define filters, options and fields
    filters = {"name": "NAME"}  # any {[key: string]: any;}
    options = {
        "limit": 100
    }  # { sort?: { [key: string]: number },limit?: number,offset?: number}
    fields = {
        "name": 1
    }  # The function will return only the required field. The rest is populated with None

    batteries = api.get_batteries(api_options=ApiOptions(filter=filters, option=options, fields=fields))
    print("My batteries:", batteries)
except:
    raise Exception(f"It was not possible to fetch your battery.")
